version: '3.8'

services:
  db:
    image: postgres:15
    container_name: genetics_db
    environment:
      POSTGRES_DB: genetics
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
    volumes:
      - postgres_data:/var/lib/postgresql/data
      - ./schema.sql:/docker-entrypoint-initdb.d/schema.sql
    ports:
      - "5434:5432"
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 10s
      timeout: 5s
      retries: 5
    networks:
      - genetics_network

  processor:
    build: 
      context: .
      dockerfile: Dockerfile.processor
    container_name: genetics_processor
    volumes:
      - .:/app:cached
    environment:
      - DB_HOST=db
      - DB_NAME=genetics
      - DB_USER=postgres
      - DB_PASSWORD=postgres
      - PYTHONPATH=/app
    depends_on:
      db:
        condition: service_healthy
    networks:
      - genetics_network
    # command: ["python", "-m", "unittest", "discover", "-s", "tests"]

networks:
  genetics_network:
    name: genetics_network

volumes:
  postgres_data:
    name: genetics_db_data